src/server/NetworkingServer.cpp

```diff
@@ -38,37 +79,19 @@ void NetworkingServer::RealtimeServer() {
         if (command_packet >> client_id >> realtime_command >> tick) {
             switch(realtime_command) {
                 case (sf::Uint32)RealtimeCommand::move:
-                    float x_dir;
-                    float y_dir;
-                    if (command_packet >> x_dir >> y_dir) {
-                        std::cout << "Client ID, tick, command, x, y: " << client_id << " " << tick << " " << realtime_command << " " << x_dir << " " << y_dir << std::endl;
-                        int drift = std::abs((int)((mCurrTick - tick)));
-                        if (drift < CMD_DRIFT_THRESHOLD) {
-                            // your newest commands will overwrite old ones
-                            std::unordered_map<sf::Uint32, float*>::const_iterator iter = mClientMoves.find(client_id);
-                            if (mAcceptingMoveCommands) {
-                                // if entry exists, don't allocate new array.
-                                if(iter != mClientMoves.end()) {
-                                    iter->second[0] = x_dir;
-                                    iter->second[1] = y_dir;
-                                }
-                                // drop the move command if we're computing game state in the other thread. doesn't feel great.
-                                mClientMoves[client_id] = new float[2]{x_dir, y_dir};
-                            } else {
-                                std::cout << "Dropping move command because game state being computed" << std::endl;
-                            }
-                        } else {
-                            std::cout << "Receive move command with tick drifted past drift threshold. Drift value is: " << drift << std::endl;
-                        }
-                    }
+                    Move(command_packet, client_id, tick);
                     break;
                 case (sf::Uint32)RealtimeCommand::fetch_state:
-                    std::cout << "Sending game state to client: " << sender << " " << port << std::endl;
+                    //std::cout << "Sending game state to client: " << sender << " " << port << std::endl;
                     // sample current state every 100 ms, this simply packages and returns it
                     game_state_packet << ct; // should have error handling for <<
-                    for (const auto iter : mClientPositions) {
-                        sf::Uint32 client_id = iter.first;
-                        game_state_packet << client_id << iter.second[0] << iter.second[1];
+                    if (mAcceptingNetworkGameObjectsReads) {
+                        for(const auto network_game_object: mNetworkGameObjects) {
+                            game_state_packet << network_game_object.id
```

Comment:
shouldn't this be simply `game_state_packet << network_game_object`?

Reply:
To achieve this, I think you need extend packets to support your custom types: https://www.sfml-dev.org/tutorials/2.5/network-packet.php